simplemodels/fields.py

```diff
@@ -242,62 +219,71 @@ class ExtraField(SimpleField):
+        return super(FloatField, self)._typecast(value, float, **{})
 
 
 class DecimalField(SimpleField):
-    def __init__(self, **kwargs):
-        self._set_default_validator(Decimal, kwargs)
-        super(DecimalField, self).__init__(**kwargs)
+    def _typecast(self, value, **kwargs):
+        return super(DecimalField, self)._typecast(value, Decimal, **{})
 
 
 class CharField(SimpleField):
+    def _typecast(self, value, **kwargs):
+        return super(CharField, self)._typecast(value, self._caster, **{})
+
     def __init__(self, is_unicode=True, max_length=None, **kwargs):
         if PYTHON_VERSION == 2:
-            validator = unicode if is_unicode else str
+            self._caster = unicode if is_unicode else str
         else:
-            validator = str
+            self._caster = str
 
-        self._set_default_validator(validator, kwargs)
+        super(CharField, self).__init__(**kwargs)
 
         # Add max length validator
-        if max_length:
-            def validate_max_length(value):
-                if len(value) > max_length:
-                    raise ValidationError(
-                        'Max length is exceeded ({} < {}) for the '
-                        'field {!r}'.format(len(value), self.max_length, self))
-                return value
-
-            self.max_length = max_length
-            self._set_default_validator(
-                validator=validate_max_length,
-                kwargs=kwargs)
+        self._max_length = max_length
+        if self._max_length is not None:
+            self._add_validator(self.validate_max_length)
 
-        super(CharField, self).__init__(**kwargs)
+    def validate_max_length(self, value):
+        if value and len(value) > self._max_length:
+            raise ValidationError(
+                'Max length is exceeded ({} < {}) for the '
+                'field {!r}'.format(len(value), self._max_length, self))
+        return True
 
-    def __set_value__(self, instance, value, **kwargs):
-        """Override method. Forbid to store None for CharField, because it
-        result to conflicts like str(None) --> 'None'
 
-        """
-        if value is None:
-            value = ''
-        return super(CharField, self).__set_value__(instance, value, **kwargs)
+class BooleanField(SimpleField):
+    def _typecast(self, value, **kwargs):
+        return super(BooleanField, self)._typecast(value, bool, **{})
 
 
-class BooleanField(SimpleField):
-    def __init__(self, **kwargs):
-        self._set_default_validator(bool, kwargs)
-        super(BooleanField, self).__init__(**kwargs)
+class DateTimeField(SimpleField):
+    DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'
+
+    def __init__(self, date_fmt=None, **kwargs):
+        self._date_fmt = date_fmt or self.DATE_FORMAT
+        super(DateTimeField, self).__init__(**kwargs)
+
+    def _typecast(self, value, **kwargs):
+        if isinstance(value, six.string_types):
+            func = partial(
```

Comment:
Partial is redundant here. 
`func = lambda val: datetime.strptime(val, self._date_fmt)` does the same thing